#### Практическое задание: Урок 3. Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных


##### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [1]:
import numpy as np
import pandas as pd

In [2]:
Salary = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

In [3]:
mean_salary = Salary.sum()/len(Salary)
mean_salary

65.3

##### Cреднее арифметическое выборки выпускников равно 65.3

Найдем среднее квадратичное отклонение для выборки (S):

In [5]:
A = ((Salary - Salary.mean())**2).sum()
print(A)
S2 = A / (len(Salary))
print(S2)
S = np.sqrt(S2)
print(S)

19002.2
950.11
30.823854398825596


##### Cреднее квадратичное отклонение для выборки (S) равно 30.823854398825596

Смещенная оценка дисперсии равна среднему квадратичному отклонению, возведенному в квадрат:

In [6]:
Var = S2
Var

950.11

Для нахождения НЕсмещенной оценки дисперсии делим сумму квадратов разностей всех величин со средним значением не на  n , а на  n−1:

In [7]:
Var2 = A / (len(Salary) - 1)
Var2

1000.1157894736842

##### Смещенная оценка дисперсии равна 950.11, НЕсмещенная - 1000.1157894736842

##### 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [9]:
from math import factorial

In [12]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Возможные варианты 3 белых мячей из выбранных 6 мячей, вытащенных из 2 ящиков:

Вариант №1: 1 ящик - 2 белых, 2 ящик - 1 белый

In [23]:
P11 = combinations(5, 2)/ combinations(8, 2)
print(P11)
# для 2 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 1 белый 2. вытащили 3 НЕбелых
P12 = combinations(5, 1)/combinations(12, 1) * combinations(7, 3)/combinations(12, 3)
print(P21)
P1 = P11*P12
print(P1)

0.35714285714285715
0.0012626262626262627
0.023674242424242424


Вариант №2: 1 ящик - 0 белых, 2 ящик - 3 белых

In [27]:
P21 = combinations(3, 2)/ combinations(8, 2)
print(P21)
# для 2 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 3 белых 2. вытащили 1 НЕбелый
P22 = combinations(5, 3)/combinations(12, 3) * combinations(7, 1)/combinations(12, 1)
print(P22)
P2 = P21*P22
print(P2)

0.10714285714285714
0.026515151515151516
0.002840909090909091


Вариант №3: 1 ящик - 1 белый, 2 ящик - 2 белых

In [29]:
# для 1 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 1 белый 2. вытащили 1 НЕбелый
P31 = combinations(3, 1)/combinations(12, 1) * combinations(5, 1)/combinations(12, 1)
print(P31)
# для 2 ящика нужно вычислить совместную вероятность 2 событий: 1. вытащили 2 белых 2. вытащили 2 НЕбелых
P32 = combinations(5, 2)/combinations(12, 2) * combinations(7, 2)/combinations(12, 2)
print(P32)
P3 = P31*P32
print(P3)

0.10416666666666667
0.048209366391184574
0.005021808999081727


Считаем совокупную вероятность 3 вариантов событий:

In [30]:
P = P1+P2+P3
P

0.03153696051423324

##### Ответ: вероятность того, что 3 мяча белые равна 0.03153696051423324

##### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Вероятность выстрела каждого из трех спортсменов равна 1/3.
Следовательно, вероятность того, что удачный выстрел произведен:

In [33]:
P1 = 1/3*0.9 + 0*0.8 + 0*0.6
P2 = 0*0.9 + 1/3*0.8 + 0*0.6
P3 = 0*0.9 + 0*0.8 + 1/3*0.6
print('вероятность того, что выстрел произведен: a). первым спортсменом равна ', P1)
print('вероятность того, что выстрел произведен: б). вторым спортсменом равна ', P2)
print('вероятность того, что выстрел произведен: в). третьим спортсменом равна ', P3)

вероятность того, что выстрел произведен: a). первым спортсменом равна  0.3
вероятность того, что выстрел произведен: б). вторым спортсменом равна  0.26666666666666666
вероятность того, что выстрел произведен: в). третьим спортсменом равна  0.19999999999999998


#### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

In [34]:
# вероятность того, что сдавший студент с факультета:
pfA = 1/4 # факультет А
pfB = 1/4 # факультет В
pfC = 2/4 # факультет С

# следовательно:
psfA = 1/4*0.8 + 0*0.7 + 0*0.9
psfB = 0*0.8 + 1/4*0.7 + 0*0.9
psfC = 0*0.8 + 0*0.7 + 2/4*0.9

print('вероятность того, что сдавший сессию студент учится на  факультете A равна ', psfA)
print('вероятность того, что сдавший сессию студент учится на  факультете В равна ', psfB)
print('вероятность того, что сдавший сессию студент учится на  факультете C равна ', psfC)

вероятность того, что сдавший сессию студент учится на  факультете A равна  0.2
вероятность того, что сдавший сессию студент учится на  факультете В равна  0.175
вероятность того, что сдавший сессию студент учится на  факультете C равна  0.45


##### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

a) совместная вероятность 3 событий - все детали выйдут из строя:

In [35]:
P = 0.1*0.2*0.25
P

0.005000000000000001

б) только две детали

In [38]:
# 1 вариант: выходят из строя 1 и 2 деталь:
P1 = 0.1*0.2
# 2 вариант: выходят из строя 1 и 3 деталь:
P2 = 0.1*0.25
# 3 вариант: выходят из строя 2 и 3 деталь:
P3 = 0.2*0.25
P = P1+P2+P3
P

0.095

в) хотя бы одна деталь

In [39]:
P = 0.1+0.2+0.25
P

0.55

г) от одной до двух деталей

In [41]:
# это совместная вероятность выхода из строя 1 детали и 2 деталей
# вероятность выхода из строя 1 детали:
P1 = 1/3*0.1+1/3*0.2+1/3*0.25
# вероятность выхода из строя 2 деталей посчитана выше
P2 = 0.095
#совместная вероятность выхода из строя 1 детали и 2 деталей:
P = P1+P2
P

0.2783333333333333